cpu/cc2538/periph/uart.c

```diff
@@ -276,28 +160,21 @@ static int init_base(uart_t uart, uint32_t baudrate)
     /* UART Enable */
     u->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 
+    /* register callbacks */
+    uart_ctx[uart].rx_cb = rx_cb;
+    uart_ctx[uart].arg = arg;
+
+    /* enable RX interrupt */
+    NVIC_EnableIRQ(UART_IRQ(uart_num));
```

Comment:
Just noticed something else: the RX part of the UART driver **must** only be initialized, if the callback parameter is not `NULL`. See the documentation in `drivers/periph/uart.h`:
>  If no callback parameter is given (rx_cb := NULL), the UART will be
>  initialized in TX only mode

So we need to factor out all the RX related code in something like
```c
if (rx_cb) {
    uart_ctx[uart].rx_cb = rx_cb;
    ...
    RXEN = 1
    ...
    NVIC_Enable ...
```

Reply:
Unrelated to this PR: but we should also consider how we include this in our testing...

Reply:
Ok. Rather than put if (rx_cb) all over the place, I've only made this control whether we enable/disable the Rx interrupt. Same reason as above. This would satisfy the statement in drivers/periph/uart.h. Was also thinking about whether to remove the rx pin configuration in case there was a conflict but reckoned that this would be up to the user of the driver to resolve...

In terms of testing, rx_cb is always defined in the current tests/periph_uart... we could provide another argument to cmd_init which initialises the driver with or without the callback function defined...?

Reply:
NB that in the commits I've cherry picked your updated uart test application. Also, I've added clock on/off tests to them. I needed to change include/uart.h though for this...